src/foam/nanos/auth/CachedUserAndGroupAuthService.java

```diff
@@ -0,0 +1,74 @@
+package foam.nanos.auth;
+
+import foam.core.X;
+import foam.nanos.util.LRULinkedHashMap;
+import java.util.Map;
+
+/**
+ * Created by marcroopchand on 2017-05-30.
+ */
+public class CachedUserAndGroupAuthService
+  extends UserAndGroupAuthService
+{
+  /**
+   * The cached data structure will look like this
+   *{
+   *  user1: {
+   *    permission1: true,
+   *    permission2: false
+   *    ...
+   *  },
+   *  user2: {
+   *    permission1: false,
+   *    permission2: true
+   *    ...
+   *  }
+   *  ...
+   *}
+   */
+  protected Map<String, Map<String, Boolean>> userMap = new LRULinkedHashMap<>(1000000);
+  protected Map<String, Boolean> permissionMap;
+
+  @Override
+  public void start() {
```

Comment:
I assumed that we will need to start this service from Boot.

How do you envision this caching fitting into the system ?